google/resource_kms_crypto_key.go

```diff
@@ -0,0 +1,187 @@
+package google
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform/helper/schema"
+	"google.golang.org/api/cloudkms/v1"
+	"log"
+	"regexp"
+	"strings"
+)
+
+func resourceKmsCryptoKey() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceKmsCryptoKeyCreate,
+		Read:   resourceKmsCryptoKeyRead,
+		Delete: resourceKmsCryptoKeyDelete,
+		Importer: &schema.ResourceImporter{
+			State: resourceKmsCryptoKeyImportState,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"location": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"key_ring": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"project": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+			},
+		},
+	}
+}
+
+type kmsCryptoKeyId struct {
+	Project  string
+	Location string
+	KeyRing  string
+	Name     string
+	Purpose  string
+}
+// TODO: Add the info about rotation frequency and start time.
+
+func (s *kmsCryptoKeyId) cryptoKeyId() string {
+	return fmt.Sprintf("projects/%s/locations/%s/keyRings/%s/cryptoKeys/%s", s.Project, s.Location, s.KeyRing, s.Name)
+}
+
+func (s *kmsCryptoKeyId) parentId() string {
+	return fmt.Sprintf("projects/%s/locations/%s/keyRings/%s", s.Project, s.Location, s.KeyRing)
+}
+
+func (s *kmsCryptoKeyId) terraformId() string {
+	return fmt.Sprintf("%s/%s/%s/%s", s.Project, s.Location, s.KeyRing, s.Name)
+}
+
+func resourceKmsCryptoKeyCreate(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
```

Comment:
If this can only be a `*Config` I recommend changing the arg type of meta. Otherwise this should probably be switched to use the non panic version of type assertion. 
```
foo, ok := bar.(Type)
if !ok { return Errors.New("Blah") }
```